app/src/main/java/com/example/myapplication/ui/LoginFragment.kt

```diff
@@ -81,14 +82,24 @@ class LoginFragment : Fragment() {
     private fun validEmailObserver() {
         loginViewModel.validEmail.observe(viewLifecycleOwner) {
             if (it == true) {
-                this.findNavController().navigate(
-                    LoginFragmentDirections.actionLoginFragmentToHomePageFragment()
-                )
+                // this.findNavController().navigate(
```

Comment:
also, you don't have to remove the navigation logic. You just need to do as follow: 
- Validate that the email and password has a correct format (you don't want to waste a backend call when you already know it will fail!)
- Afterwards, when the format validation observable notifies that the email its okay, you should send the request, and the response observable know holds the responsibility to navigate if the response its okay (or show a error toast if so).